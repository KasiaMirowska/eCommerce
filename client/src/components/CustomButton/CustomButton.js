import React from 'react';
//import './CustomButton.styles.scss';
import {CustomButtonContaner} from './CustomButton.styles';

// export default function CustomButton({children, isGoogleSignedIn, invertedColor, ...otherProps}) {
//     //conditionally rendering a button class based on a prop for google signIn...if google? then it has a google class , if not then nothing plus always custom-button class...

//     return (
//         <button className={`${isGoogleSignedIn? 'google-sign-in' : ''} ${invertedColor? 'invertedColor' : ''} custom-button`} {...otherProps}>
//             {children}
//         </button>
//     )
// }


export default function CustomButton(props) {
    //after creating styled component this becomes simpler then above - we put rendering conditionals into the styled component

    return (
        <CustomButtonContaner {...props} >
            {props.children}
        </CustomButtonContaner>
    )
}