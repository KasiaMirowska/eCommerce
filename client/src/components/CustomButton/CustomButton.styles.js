import styled, {css} from 'styled-components';

//instead of conditionally rendering classes in the component to implement different rendering needs we will condionally render css bloks here inside the styled component
const getButtonStyles = props => {
    if(props.googleSignInS) {
        return googleSignInStyles;
    } 
    return props.inverted? invertedButtonStyles : buttonStyles;
}


const invertedButtonStyles = css `
    background-color: white;
    color: black;

    &:hover {
    background-color:black;
    color: white;
    }
`

const googleSignInStyles = css `
    background-color: #4285f4;
    color: white;

    &:hover {
        background-color:#357ae8;
        border: none;
    }
`

const buttonStyles = css `
    background-color: black;
    color: white;

    &:hover {
        background-color: rgb(210, 250, 81);
        color: black;
        }
`
export const CustomButtonContaner = styled.button `
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 35px 0 35px;
    font-size: 15px;
    text-transform: uppercase;
    font-family: 'Open Sans Condensed';
    font-weight: bolder;
    border: none;
    cursor: pointer;
    display:flex;
    justify-content: center;

    ${getButtonStyles} 
    // calling the above function determining which style to render

`