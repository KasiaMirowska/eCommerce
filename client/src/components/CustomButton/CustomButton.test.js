import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import CustomButton from '../../components/CustomButton/CustomButton';


describe('CustomButton', () => {
    const props = {};

    const wrapper = shallow(<CustomButton {...props} />);

    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
        const div = document.createElement('div');
        const item = renderer.create(<CustomButton {...props}/>, div);
        expect(item.toJSON()).toMatchSnapshot();
    });
})