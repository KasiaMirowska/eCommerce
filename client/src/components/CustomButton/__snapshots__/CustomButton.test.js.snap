// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders as expected 1`] = `ShallowWrapper {}`;

exports[`CustomButton renders as expected 2`] = `
<button
  className="sc-AxjAm dymsBa"
/>
`;
